homeassistant/components/coolmaster/config_flow.py

```diff
@@ -0,0 +1,44 @@
+"""Config flow to configure Coolmaster."""
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PORT
+import homeassistant.helpers.config_validation as cv
+
+from .const import AVAILABLE_MODES, CONF_SUPPORTED_MODES, DEFAULT_PORT, DOMAIN
+
+MODES_SCHEMA = {vol.Required(mode): bool for mode in AVAILABLE_MODES}
+HOST_SCHEMA = {
+    vol.Required(CONF_HOST): str,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+}
+
+DATA_SCHEMA = vol.Schema({**HOST_SCHEMA, **MODES_SCHEMA})
```

Comment:
Just inline HOST_SCHEMA in here ?

Reply:
That's how I had it originally, but it didn't look good. Python is not a language I'm proficient with, so happy to rewrite this in a more idiomatic way if one exists.